### DISCUSSION TITLE
Encryption algorithm config for Juniper.

### DISCUSSION THREAD

**Discussion by skvedo** [strongswan/strongswan#883] [Q&A] [answered]
Hello all,

I updating one deployment where strongSwan is one side in L2L IPSEC against Juniper SRX345. I have issue correctly specify encryption algorithms  and getting NO_PROPOSAL_CHOSEN

Juniper side:
Phase1: 
```
security {
ike {
proposal <VPN> {
    authentication-method pre-shared-keys;
    dh-group group19;
    encryption-algorithm aes-256-gcm;
}
}
}
Phase 2: 
security {
ipsec {
proposal <PROP-VPN> {
    protocol esp;
    encryption-algorithm aes-256-gcm;
    lifetime-seconds 3600;
}
policy <POL-VPN> {
    perfect-forward-secrecy {
        keys group19;
    }
    proposals <PROP-VPN>;
}
}
}
```

Old config in /etc/ipsec.d/<vpn.conf>
```
config setup
    protostack=netkey

conn <vpn1>
     also=<vpn2>
     leftsubnet=172.26.0.0/24
     rightsubnet=172.31.0.0/16
     auto=start

conn <vpn2>
     ikev2=insist
     leftid=<myIP>
     right=<srxIP>
     authby=secret
     ike=aes_gcm_c-sha384;dh19
     phase2=esp
     phase2alg=aes_gcm256-null
     salifetime=3600s
```

My config in /etc/swarmctl/swanctl.conf
```
connections {

   <srx> {
      remote_addrs = <srxIP>

      local {
         auth = psk
         id = <mypublicIP>
      }
      remote {
         auth = psk
         id = <srxIP>
      }
      children {
         net-net {
            local_ts  = <localSubnet>
            remote_ts = <remotesubnet>

            updown = /usr/local/libexec/ipsec/_updown iptables
            rekey_time = 5400
            rekey_bytes = 500000000
            rekey_packets = 1000000
            esp_proposals = default
         }
      }
      version = 2
      mobike = no
      reauth_time = 10800
      proposals = default
   }
}

secrets {
   ike-1 {
      id-me = <mypublicIP>
      id-srx = <srxIP>
      secret = <pass>
   }
 
}
```

When I use proposals/esp_proposals same values from previous config, it is not parsed.
```
root@c13a3c31ff1b:/etc/swanctl# swanctl --load-all
loaded ike secret 'ike-1'
no authorities found, 0 unloaded
no pools found, 0 unloaded
loading connection '<srx>' failed: invalid value for: proposals, config discarded
loaded 0 of 1 connections, 1 failed to load, 0 unloaded
```
Tried multiple variants of config, but can't find any working one.
Can someone bump me into right direction please.
Than you very much.

**Comment by tobiasbrunner** [comment, marked as answer]
> When I use proposals/esp_proposals same values from previous config, it is not parsed.

Because that syntax (with `;` separating DH groups) is not from strongSwan but some other *swan variant.

The default IKE proposals should be fine as the AEAD proposal will include `aes256gcm16`, `prfsha384` and `ecp256` (the latter requires the _openssl_ plugin, or some other third-party crypto provider), but you could configure it explicitly: `proposals = aes256gcm16-prfsha384-ecp256`.

For the ESP proposal you want to use `esp_proposals = aes256gcm16-ecp256` explicitly as the default proposals don't include DH groups.

**Reply by skvedo** [reply]
Thanks Tobias,

config loads, but from packet capture I see like it still sends same proposal.
![image](https://user-images.githubusercontent.com/29920587/152763504-e57932b3-e3da-4ab4-a0bc-ebf156fb7a3a.png)

Michal

**Reply by tobiasbrunner** [reply]
You are looking at the wrong payload. That payload is to propose hash algorithms used for signature authentication.

**Reply by skvedo** [reply]
My bad. Thanks. Still not working but I will try to capture ike from old deployment and compare with this one.

**Reply by tobiasbrunner** [reply]
You still receive `NO_PROPOSAL_CHOSEN`? Any logs from the Juniper end?

**Reply by skvedo** [reply]
Juniper was still responding with NO_PROPOSAL_CHOSEN and didn't find more details in debuging.
Changed configuration a bit with intention initiate connection from Juniper and compare packet captures. Realized I didn't had tunnel interface in security zone. After adding it to zone, Juniper started cooperating and tunnel established.

Thanks for your help with prosasals.
